app/src/main/java/com/babylon/wallet/android/presentation/dapp/authorized/login/DAppAuthorizedLoginViewModel.kt

```diff
@@ -193,29 +193,42 @@ class DAppAuthorizedLoginViewModel @Inject constructor(
                 }
                 return Result.success(Unit)
             } else {
-                return Result.failure(RadixWalletException.DappRequestException.InvalidPersonaOrAccounts)
+                return Result.failure(InvalidPersonaOrAccounts)
             }
         }
         return Result.success(Unit)
     }
 
+    @Suppress("ReturnCount")
     private suspend fun validateRequestedAddressesForProofOfOwnership(
         requestedPersonaAddress: IdentityAddress?,
         requestedAccountAddresses: List<AccountAddress>?
     ): Boolean {
-        // if requestedPersonaAddress is present, then check if it exists in the allPersonasOnCurrentNetwork
+        // if both requestedPersonaAddress and requestedAccountAddresses are null then return false
+        if (requestedPersonaAddress == null && requestedAccountAddresses == null) return false
+
+        val profile = getProfileUseCase()
+
+        // validate requestedPersonaAddress if present
         val isIdentityValid = requestedPersonaAddress?.let { address ->
-            getProfileUseCase().activePersonasOnCurrentNetwork.any { it.address == address }
-        } ?: true // if requestedPersonaAddress is null, assume it's valid (no need to check)
+            profile.activePersonasOnCurrentNetwork.any { it.address == address }
+        } ?: true // if requestedPersonaAddress is null then it's considered valid
```

Comment:
So here can return false immediately?

Below it is stated:
```
// if requestedPersonaAddress is not present then return false
if (!isIdentityValid) return false
```

Reply:
no because then we would never check the accounts